src/containers/DataTable/index.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+
+import Pagination from '../../components/Pagination';
+import Table from '../../components/Table';
+import Search from '../../components/Search';
+import {
+  calculateTotalNumberOfPages,
+  getSearchResult,
+  getCurrentPageData,
+} from './helpers';
+
+const DataTable = ({ rows, rowsPerPage = 40 }) => {
+  const [currentPageNumber, setCurrentPageNumber] = useState(0);
+  const [queryString, setQueryString] = useState('');
+
+  const filteredRows = getSearchResult(rows, queryString);
```

Comment:
All the needed data could be derived from props and state, so we could have the minimal state